etl/tests/test_flask_app.py

```diff
@@ -48,19 +70,34 @@ def test_robots(test_client: testing.FlaskClient) -> None:
     assert get_return.status_code == HTTPStatus.NOT_FOUND
 
 
+@pytest.mark.timeout(10)
 @pytest.mark.usefixtures('populated_azure_emulator')
 def test_run_tl(test_client: testing.FlaskClient,
+                thread_runner: flask_app.ThreadRunner,
                 mongo_client: pymongo.MongoClient,
                 database_name: str,
                 collection: str,
                 sample_salt: str,
                 sample_signature: str) -> None:
 
     post_return = test_client.post('/run_tl', data=dict(salt=sample_salt, signature=sample_signature))
-    assert post_return.status_code == HTTPStatus.CREATED
+    assert post_return.status_code == HTTPStatus.OK
+    while thread_runner.is_thread_running():
+        time.sleep(0.1)
     assert mongo_client[database_name][collection].count() == 7
 
 
+@pytest.mark.timeout(10)
+@pytest.mark.usefixtures('populated_azure_emulator', 'thread_runner')
```

Comment:
Why does this test need to use the `thread_runner` fixture?

Reply:
To ensure that the Thread is torndown (ie waits until the thread is done) after the test finishes.